### ISSUE TITLE
REF-001: Add option to exclude paths or directories from link validation

### ISSUE THREAD

**Issue by nozomi-koborinai** [nozomi-koborinai/contextlint#18]
## Problem

REF-001 flags relative links that point to files outside the current repository (e.g., a sibling reference repository or files not yet created). These are expected to be unresolvable in the lint context but are valid in the project's runtime environment.

Examples:
- `../../_references/other-repo/docs/spec.md` — sibling repository, not cloned in CI
- `../../../prisma/seed/setup.ts` — file planned but not yet created

Currently there is no way to suppress these without disabling REF-001 entirely.

## Proposal

Add an optional `exclude` (glob or prefix array) parameter to REF-001:

```jsonc
{
  "rule": "ref001",
  "options": {
    "exclude": ["_references/**", "prisma/**"]
  }
}
```

Links matching any exclude pattern would be skipped. This keeps REF-001 useful for catching genuinely broken links while avoiding false positives from external references.

## Environment

- contextlint v0.2.0
